src/app/tarea-dialog/tarea-dialog.component.ts

```diff
@@ -19,18 +22,30 @@ export class TareaDialogComponent implements OnInit, OnDestroy {
   constructor(private formBuild: FormBuilder,
     private tareaSvc: TareaService,
     private _snackBar: MatSnackBar,
-    private spinner: SpinnerService) { }
+    private spinner: SpinnerService,
+    public dialog:MatDialog,
+    @Inject(MAT_DIALOG_DATA) public data:Tarea
+  ) { }
 
   ngOnInit(): void {
-    this.formTarea = this.formBuild.group({
-      titulo: ['', [Validators.required, Validators.maxLength(30), Validators.pattern(/^[a-zA-Z 0-9.]+$/)]],
-      description: ['', Validators.maxLength(250)]
 
-    });
+      this.formTarea = this.formBuild.group({
+        titulo: ['', [Validators.required, Validators.maxLength(30), Validators.pattern(/^[a-zA-Z 0-9.]+$/)]],
+        description: ['', Validators.maxLength(250)]
+      });
 
+      if(this.data)
+        this.formTarea.patchValue(this.data);
```

Comment:
no pases todo el objeto data en el patchValue, creo que puedes hacer algo como esto 
patchValue({
titulo: this.data.titulo,
description: this.data.description,
});
asi tienes mas control de lo que estas pasando a patchValue